src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -246,7 +257,11 @@ var OptionalPortsAndServiceDiscoveryComponent = React.createClass({
   },
 
   render: function () {
-    if (isTooComplexStructure(this.props.fields)) {
+    var hasPlaceIdSet = PluginComponentStore.hasComponentAtPlaceId(
+      PluginMountPoints.OPTIONAL_PORTS_AND_SERVICE_DISCOVERY
+    );
+
+    if (!hasPlaceIdSet && isTooComplexStructure(this.props.fields)) {
```

Comment:
So we're guaranteed to be able to parse the ports as long as a plugin is mounted here?